scripts/mcp_patch.diff

```diff
@@ -2390,17 +2390,29 @@ diff -u1 -rN MCP-Reborn-vanilla/src/main/java/com/minerl/multiagent/env/EnvServe
 +    }
 +
 +    public static void execActions(String actions, int options) {
++        doChatActions(actions);
 +        KeyboardListener.State keysState = constructKeyboardState(actions);
 +        MouseHelper.State mouseState = constructMouseState(actions);
 +        PlayRecorder.getInstance().setMouseKeyboardState(mouseState, keysState);
 +        ReplaySender.getInstance().addAction(mouseState, keysState);
 +    }
++    
++    
++    public static void doChatActions(String actions) {
++        for (String action: actions.split("\n")) {
++           String[] splitAction = action.trim().split(" ");
++           if (splitAction[0].equals("chat") && splitAction.length > 1) {
++              String chatMessage = action.trim().substring(splitAction[0].length()).trim();
++              Minecraft.getInstance().player.sendChatMessage(chatMessage);
++           }
++        }
++    }
 +
 +    private static KeyboardListener.State constructKeyboardState(String actions) {
 +        List<String> keysPressed = new ArrayList<>();
 +        for (String action: actions.split("\n")) {
 +           String[] splitAction = action.trim().split(" ");
-+           if (!splitAction[0].equals("camera") && !splitAction[0].equals("dwheel")) {
++           if (!splitAction[0].equals("camera") && !splitAction[0].equals("dwheel") && !splitAction[0].equals("chat")) {
```

Comment:
this an the check below are needed to prevent parseInt from throwing an exception when given a string and crashing the server